### PR TITLE
Setup depedabot with auto-merge

### PR DESCRIPTION
**Pull request by janisz**
This PR configures depedabot for this repo and auto-merge it's PRs if all checks passes.
This is needed as old dependencies in this projects cause failures on projects that depends on it when they try to update. 
- https://github.com/stackrox/stackrox/pull/6174#issuecomment-1567193736

Configuration is similar to one done in other stackrox repositories.

Tested [here](https://github.com/stackrox/helmtest/actions/runs/5123643334/jobs/9214446347?pr=28#step:3:11) but PR needs to be created by dependabot :(

### CODE REVIEW

File: .github/workflows/auto-merge.yml
Comment (janisz) [inline .github/workflows/auto-merge.yml:13]: @SimonBaeumer since I have no rights to this repo could you add this secret?

Comment (GrimmiMeloni) [reply .github/workflows/auto-merge.yml:13]: Added you @janisz (and Draco and Automation) as Admins.

Comment (janisz) [reply .github/workflows/auto-merge.yml:13]: I've added secret

File: .github/dependabot.yml
Comment (dhaus67) [inline .github/dependabot.yml:22]: Screenshot 2023-05-31 at 05 48 09

**Review by dhaus67 — APPROVED** [review]
LGTM, one nit regarding the newline.

I suppose we should wait for one approval from Draco to check whether they are OK with the CODEOWNERS change.

**Review by vladbologa — APPROVED** [review]

**Pull request merged**
